frontend/app/routes/$lang/_protected/applications/$id.tsx

```diff
@@ -83,18 +83,63 @@ export async function loader({ context: { session }, params, request }: LoaderFu
 
   const allFederalSocialPrograms = lookupService.getAllFederalSocialPrograms();
   const allProvincialTerritorialSocialPrograms = lookupService.getAllProvincialTerritorialSocialPrograms();
+
+  const children = applicationDetails.children?.map((child) => {
+    const selectedFederalBenefit = child.dentalBenefits?.federalSocialProgram && localizeFederalSocialProgram(lookupService.getFederalSocialProgramById(child.dentalBenefits.federalSocialProgram), locale);
+    const selectedProvincialBenefit =
+      child.dentalBenefits?.provincialTerritorialSocialProgram && localizeProvincialTerritorialSocialProgram(lookupService.getProvincialTerritorialSocialProgramById(child.dentalBenefits.provincialTerritorialSocialProgram), locale);
+    console.log(' ');
+    console.log(' ');
+    console.log(' ');
+    console.log(' ');
+    console.log(selectedFederalBenefit);
+    console.log(selectedProvincialBenefit);
+    console.log(' ');
+    console.log(' ');
+    console.log(' ');
+    console.log(' ');
+
+    return {
+      firstName: child.information.firstName,
+      lastName: child.information.lastName,
+      birthday: child.information.dateOfBirth,
+      sin: child.information.socialInsuranceNumber ? formatSin(child.information.socialInsuranceNumber) : '',
+      isParent: child.information.isParent,
+      dentalInsurance: {
+        acessToDentalInsurance: child.dentalInsurance,
+        federalBenefit: {
+          hasFederalBenefits: child.dentalBenefits?.hasFederalBenefits,
+          benefit: selectedFederalBenefit != '' ? selectedFederalBenefit?.name : undefined,
+        },
+        provincialTerritorialBenefit: {
+          hasProvincialTerritorialBenefits: child.dentalBenefits?.hasProvincialTerritorialBenefits,
+          province: child.dentalBenefits?.province,
+          benefit: selectedProvincialBenefit != '' ? selectedProvincialBenefit?.name : undefined,
```

Comment:
Hard to tell from only the changes in this file alone but could `selectedProvincialBenefit` actually be an empty string here? Line 88 above indicates that this variable is initialized using calls to a utility function, which shouldn't return an empty string in my opinion.

Same comment for `selectedFederalBenefit` above.

Reply:
I had to add that check for both fields because I was getting the following error otherwise:
```
Property 'name' does not exist on type '"" | { name: string; id: string; provinceTerritoryStateId: string; }'.
  Property 'name' does not exist on type '""'
```

Reply:
After some discussion, fixing it as suggested: 2d8355e9ebd14201a9ab355ac64a92d96c3b4b83.